packages/smooth_app/lib/pages/product/add_basic_details_page.dart

```diff
@@ -0,0 +1,185 @@
+                    ),
+                    SizedBox(height: size.height * 0.02),
+                    SmoothTextFormField(
+                      controller: _brandNameController,
+                      type: TextFieldTypes.PLAIN_TEXT,
+                      hintText: appLocalizations.brand_name,
+                      validator: (String? value) {
+                        if (value == null || value.isEmpty) {
+                          return appLocalizations
+                              .add_basic_details_brand_name_error;
+                        }
+                        return null;
+                      },
+                    ),
+                    SizedBox(height: size.height * 0.02),
+                    SmoothTextFormField(
+                      controller: _weightController,
+                      type: TextFieldTypes.PLAIN_TEXT,
+                      hintText: appLocalizations.weight_gram,
+                      textInputType: TextInputType.number,
+                      validator: (String? value) {
+                        if (value == null || value.isEmpty) {
+                          return appLocalizations
+                              .add_basic_details_weight_gram_error;
+                        }
+                        return null;
+                      },
+                    ),
+                    SizedBox(height: size.height * 0.05),
+                  ],
+                ),
+              ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceAround,
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: <Widget>[
+                _buildButton(
+                  appLocalizations.cancel,
+                  () => Navigator.pop(context),
+                ),
+                _buildButton(appLocalizations.save, () {
+                  if (!_formKey.currentState!.validate()) {
+                    return;
+                  }
+                  saveData();
+                }),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void _errormessageAlert(final String message) => showDialog<void>(
+        context: context,
+        builder: (BuildContext context) => SmoothAlertDialog(
+          body: ListTile(
+            leading: const Icon(Icons.error_outline, color: Colors.red),
+            title: Text(message),
+          ),
+          actions: <SmoothActionButton>[
+            SmoothActionButton(
+              text: AppLocalizations.of(context)!.close,
+              onPressed: () => Navigator.pop(context),
+            ),
+          ],
+        ),
+      );
+
+  Future<void> saveData() async {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    widget.product.productName = _productNameController.text;
+    widget.product.quantity = _weightController.text;
+    widget.product.brands = _brandNameController.text;
+    final Status status = await OpenFoodAPIClient.saveProduct(
```

Comment:
Actually you should call `LoadingDialog`, shouldn't you?

Reply:
Yes, we should be calling `LoadingDialog` for a better user experience.